### PR TITLE
Process any native message

### PR DESCRIPTION
**Pull request by mauroservienti**

### CODE REVIEW

File: transports/sqs/native-integration_messagetypedetection_sqs_[6.2,).partial.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+* If none of the above apply, the transport tries to deserialize the message into a `TransportMessage` envelope. If it fails or there are no NServiceBus specific headers (`NServiceBus.MessageID` and `NServiceBus.EnclosedMessageTypes`), the message is treated as a native message and the message body passed as is to the processing pipeline. These is no support for loading the message body from an [S3 bucket](/transports/sqs/configuration-options.md#offload-large-messages-to-s3)
+
+Whenever the native message needs to be copied for [moving messages to the error queue](/nservicebus/recoverability), [auditing](/nservicebus/operations/auditing.md) or [delayed retries](/nservicebus/recoverability/configure-delayed-retries.md) purposes, the native message is converted into the transport's internal structure. The collection from `NServiceBus.AmazonSQS.Headers` is moved from the native message attribute into the [headers collection](/nservicebus/messaging/headers.md). All other available message attributes from the original native message are copied over into the newly formed native message.
```
Comment (mauroservienti) [inline transports/sqs/native-integration_messagetypedetection_sqs_[6.2,).partial.md:11]: I'm not sure this is true. If the endpoint is configured not to wrap messages, failed, audited, and delayed messages should not be wrapped as well.

File: transports/sqs/native-integration_messagetypedetection_sqs_[6.2,).partial.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+* If the SQS message has an header named `MessageTypeFullName`, the message is treated as a native message and the value of the `MessageTypeFullName` will be used as the type to deserialize the message. The native message body is loaded from the configured [S3 bucket](/transports/sqs/configuration-options.md#offload-large-messages-to-s3) when the message has an `S3BodyKey` attribute with the value representing an S3 object key, including the [necessary prefix](/transports/sqs/configuration-options.md#offload-large-messages-to-s3-key-prefix) as configured by the receiving endpoint.
+* If none of the above apply, the transport tries to deserialize the message into a `TransportMessage` envelope. If it fails or there are no NServiceBus specific headers (`NServiceBus.MessageID` and `NServiceBus.EnclosedMessageTypes`), the message is treated as a native message and the message body passed as is to the processing pipeline. These is no support for loading the message body from an [S3 bucket](/transports/sqs/configuration-options.md#offload-large-messages-to-s3)
```
Comment (mauroservienti) [inline transports/sqs/native-integration_messagetypedetection_sqs_[6.2,).partial.md:9]: Do we think adding support for loading the body from S3 in this scenario is worth it?

File: transports/sqs/native-integration_messagetypedetection_sqs_[6.2,).partial.md (hunk 1 above)
Comment (SzymonPobiega) [reply transports/sqs/native-integration_messagetypedetection_sqs_[6.2,).partial.md:11]: I am pretty sure it is true:) Let's try that out in a sample.

**Review by SzymonPobiega — APPROVED** [review]

File: transports/sqs/native-integration_messagetypedetection_sqs_[6.2,).partial.md [hunk 3]
```diff
@@ -0,0 +1,11 @@
+### Message type detection
```
Comment (mikeminutillo) [inline transports/sqs/native-integration_messagetypedetection_sqs_[6.2,).partial.md:1]: NOTE: This is not currently rendered as there are no 6.2 snippets

**Pull request merged**